test/Maps/MapReader.test.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Maps/MapReader.h"
+#include "Maps/MapType.h"
+#include "gtest/gtest.h"
+#include <string>
+
+TEST(MapReader, EmptyFilename) {
+	EXPECT_THROW(MapReader::Read(std::string(), MapType::Map), std::runtime_error);
+	EXPECT_THROW(MapReader::Read(std::string(), MapType::Save), std::runtime_error);
+}
+
+TEST(MapReader, MissingFile) {
+	EXPECT_THROW(MapReader::Read("MissingFile.map", MapType::Map), std::runtime_error);
+	EXPECT_THROW(MapReader::Read("MissingFile.op2", MapType::Save), std::runtime_error);
+}
```

Comment:
I had to think about this. I wasn't sure if this is really testing the `MapReader`, or if it's actually just testing the underlying `Stream::FileReader` class. I think it does make sense though, since at this level we don't know it's using an underlying `Stream::FileReader` object, and technically that is subject to change. From an API standard, we do want well defined behaviour if a bad filename is passed, and that includes throwing an exception.

Do we need to test both an empty file name, as per the test above, and a missing file name? Both are technically the same, a non-existent file. Or maybe we should test them all together in the same test. Hmm, the choice is not clear to me.

Reply:
I'm not sure it is necessary. 

While an empty string should also report as a missing file, something could break or not play nicely if using the filename for error reporting or something else where a non-empty string was expected. This shouldn't matter in the current MapReader code though.

I think for an easier test like this, it doesn't really bloat the testing documentation to include it even if it mostly overlaps what is being tested?

Reply:
Ahh, not such a bad idea to consider the error reporting like that. Ok, I'm fine with including both empty strings and non-empty strings for non-existent files.

My preference is still to combine them into a single test method, since they are so similar.

Reply:
Okay, I'll combine them.